RandBLAS/sparse.hh

```diff
@@ -234,32 +233,15 @@ static RNGState template_fill_sparse(
         vec_ax_idxs = long_ax_idxs;
         rep_ax_idxs = short_ax_idxs;
     }
-    RNGState out_state = repeated_fisher_yates<T, T_gen>(
+    S0.next_state = repeated_fisher_yates(
         S0.seed_state, num_vecs, vec_len,
         S0.dist.vec_nnz, vec_ax_idxs, rep_ax_idxs, S0.vals
     );
-    S0.next_state = out_state;
-    return out_state;
+    return S0.next_state;
 }
 
-
-template <typename T>
-RNGState fill_sparse(
-    SparseSkOp<T> &S0
-) {
-    switch (S0.seed_state.rng_name) {
-        case RNGName::Philox:
-            return template_fill_sparse<T, Philox>(S0);
-        case RNGName::Threefry:
-            return template_fill_sparse<T, Threefry>(S0);
-        default:
-            throw std::runtime_error(std::string("Unrecognized generator."));
-    }
-}
-
-
-template <typename T>
-void print_saso(SparseSkOp<T>& S0) {
+template <typename SKOP>
+void print_saso(SKOP const& S0) {
```

Comment:
Same question as before: why not just use ``SparseSkOp<T>``?

Reply:
to simplify the usage, it's tedious to have to manually specify all of the nested types when the compiler can do it for us.